netvirt.rst

```diff
@@ -0,0 +1,719 @@
+fundamental to SDN.
+
+
+The application of SDN to create virtual networks is generally
+credited to the team at Nicira, as discussed in Chapter 2, with a 2014
+paper by Koponen *et al.* laying out the architecture of an SDN-based
+system to implement network virtualization.
+
+
+.. _reading_NVP:
+.. admonition:: Further Reading
+                
+   T. Koponen et al. `Network Virtualization in Multi-tenant Datacenters
+   <https://www.usenix.org/conference/nsdi14/technical-sessions/presentation/koponen>`__.
+   NSDI, April, 2014.
+   
+
+The following sections show how a particular set of technology
+challenges, combined with the new capabilities offered by SDN, set the
+stage for network virtualization as a successful use case for
+Software-Defined Networks.
+
+8.1 Challenges
+--------------
+
+Network virtualization as we understand it today is closely linked to
+the evolution of modern data centers, in which large numbers of
+commodity servers communicate with each other to solve computational
+tasks. These data centers are common for both large cloud providers
+(e.g., AWS, Azure, Google), as well as many enterprise organizations.
+Some of the challenges involved in building networks for such data
+centers were laid out in the VL2 paper from Microsoft Research back
+in 2009.
+
+
+.. _reading_VL2:
+.. admonition:: Further Reading
+
+   Greenberg et al. `VL2: a scalable and flexible data center network
+   <https://dl.acm.org/doi/10.1145/1594977.1592576>`__.
+   SIGCOMM, August, 2009.
+
+In such data centers, there is a substantial amount of “east-west”
+traffic; that is, server-to-server traffic, as distinct from
+“north-south” traffic entering or leaving the data center. To
+efficiently support high volumes of traffic between any pair of
+servers in the data center, leaf-spine fabrics of the sort described
+in Chapter 7 became popular due to their high cross-sectional
+bandwidth and scalable layer-3 forwarding.
+
+
+At the same time, server virtualization became mainstream, which had
+several implications for data center operations. Provisioning virtual
+machines (VMs) can be carried out entirely in software, by contrast to
+installing and configuring physical servers, a time-consuming and
+manual process. As the ease of provisioning a VM shrank the time to
+obtain computational resources from days to minutes or seconds, it
+exposed the fact that network configuration was now the "long
+pole”—the slowest task to be completed before a user could put their
+infrastructure to work. Hence, there was a recognition that network
+configuration and provisioning needed to move towards a model more
+similar to virtual compute, setting the stage for network
+virtualization.
+
+
+A second effect of server virtualization was to enable virtual machine
+mobility. Since a VM has an IP address, this introduced some real
+challenges for data center networking. If a VM were to move from one
+subnet to another—an entirely reasonable thing to do—it would need to
```

Comment:
Maybe subnet is a wrong word here? The vm has an ip from an ip subnet and to get mobility that subnet needs to be individually routable in de underlay if plain ip is used. As explained below, the problem is different when L2 ethernet is used as underlay... For me this paragraph was confusing as it is written now, maybey a picture will help?

Reply:
It's a good observation. I left out some of my reasoning and the details of what happens when the VM moves. Will update with an extra paragraph.